react/features/filmstrip/middleware.web.js

```diff
@@ -53,3 +69,47 @@ MiddlewareRegistry.register(store => next => action => {
     return result;
 });
 
+/**
+ * Private helper to calculate the reordered list of remote participants when a remote participant joins.
+ *
+ * @param {*} store - The redux store.
+ * @returns {void}
+ * @private
+ */
+function _updateRemoteParticipantsonJoin(store) {
+    const state = store.getState();
+    const { remote } = state['features/base/participants'];
+
+    if (!remote) {
+        return;
+    }
+    const { remoteParticipants } = state['features/filmstrip'];
+    const currentParticipants = Array.from(remote.values());
+
+    // Sort the remote participants alphabetically.
+    const reorderedParticipants = sortRemoteParticipants(currentParticipants);
```

Comment:
Isn't this going to place the screen shares and the previous speakers on the wrong positions?